internal/handlersettings/handlersettingscommon.go

```diff
@@ -112,14 +112,27 @@ func unmarshalProtectedSettings(configFolder string, hs settings.SettingsCommon,
 	// we use os/exec instead of azure-docker-extension/pkg/executil here as
 	// other extension handlers depend on this package for parsing handler
 	// settings.
-	cmd := exec.Command("openssl", "smime", "-inform", "DER", "-decrypt", "-recip", crt, "-inkey", prv)
+
+	//using cms command to support for FIPS 140-3
+	cmd := exec.Command("openssl", "cms", "-inform", "DER", "-decrypt", "-recip", crt, "-inkey", prv)
```

Comment:
Have you verified this to be working without errors ? tested end to end on a VM ?